src/v/resource_mgmt/cpu_profiler.cc

```diff
@@ -36,6 +36,13 @@ cpu_profiler::cpu_profiler(
   , _sample_period(std::move(sample_period)) {
     _enabled.watch([this] { on_enabled_change(); });
     _sample_period.watch([this] { on_sample_period_change(); });
+
+    for (size_t i = 0; i < number_of_results_buffers; i++) {
+        _results_buffers.emplace_back(
```

Comment:
I don't see how this helps. `number_of_results_buffers` is 10 and `_results_buffers` is `std::deque<profiler_result>` and `profile_result` is 40 bytes so this is just `_results_buffers` adding 400 bytes worth of elements to `_results_buffers`?

Reply:
It's about the profile_result.samples but your point still stands as this doesn't help with that neither. 

I will need to double check tomorrow why I wasn't getting the oversized warnings anymore with this change.

Reply:
Oh I get it, because in that case the allocation happens in the reactor which just does an `assign` so in my simple testing this doesn't have 128 samples and hence the allocation is actually rather small.

Reply:
Added an `reserve` call.